lib/polymorphic_integer_type/polymorphic_foreign_association_extension.rb

```diff
@@ -3,7 +3,7 @@ module PolymorphicForeignAssociationExtension
 
     def set_owner_attributes(record)
       super
-      if reflection.foreign_integer_type && reflection.integer_type
+      if reflection.try(:foreign_integer_type) && reflection.try(:integer_type)
```

Comment:
Hmm.  My general feeling is that the use of `try` is a bit lazy => why not simply have those methods defined but return nil to that we end up with reflections continuing to look similar, versus needing to remember that we need to guard against some differences.

Reply:
you are more suggesting to something like this?
```
def foreign_integer_type(reflection)
  if reflection.respond_to? (:foreign_integer_type)
    reflection.foreign_integer_type
  else
    nil
  end  
end
```
or simple
```
def foreign_integer_type(reflection)
  reflection.try(: foreign_integer_type)
end
```
which feel like the same though

Reply:
Also, I do not want to just adding more`attr_accessor` to `ThroughReflection ` in here https://github.com/clio/polymorphic_integer_type/blob/master/lib/polymorphic_integer_type/extensions.rb#L5-L8 in case there is more reflection I am not aware of.